### ISSUE TITLE
Review if maximum DR eligibility is working properly

### ISSUE THREAD

**Issue by mariocao** [witnet/witnet-rust#1960]
In the commit bd03469652cfb6007917c83f38e66bd201b78510 of the `202106-recovery` branch, a new maximum data request eligibility was included. However, some logs were seen by @tmpolaczyk stating that the probability for mining a data request was 100%.

Some of the tasks that should be covered by this issue:
- [x] Code review of the DR maximum eligibility logic
- [x] Implement TODO at `node/src/actors/chain_manager/mining.rs:325` (pass difficulty as argument)
- [x] Implement some tests that check new DR maximum eligibility logic (after third hard-fork)
- [x] Identify and fix the problem of the mistmatch between the new maximum DR elelibility and the 100% prob. log

**Comment by tmpolaczyk** [comment]
After taking a closer look I think that the problem is that the threshold factor is sometimes set to u32::MAX. The threshold factor is calculated in this function:

https://github.com/witnet/witnet-rust/blob/9d5e6b6e8ad1605e85472bc243a183a13697b393/data_structures/src/chain.rs#L3543

The threshold factor it is set to u32::MAX when the number of witnesses in a data request is greater that the number of active identities.

This threshold is used in calculate_reppoe_threshold, but notice how the minimum difficulty is applied before multiplying by the threshold:

https://github.com/witnet/witnet-rust/blob/d8cf21857d8466b783af640c992af2e01a12c8e8/validations/src/validations.rs#L2115-L2121

`(max / u64::from(minimum_difficulty))` is `max / 2000`, but multiplied by u32::MAX it goes back to `max`.

So the minimum difficulty implementation doesn't work when the ARS is too small, which was the original problem that we were trying to solve. It does change the eligibility in some other cases, because the blocks are no longer valid after that change, but the number of commits when the ARS is small should be the same as before.

**Comment by tmpolaczyk** [comment]
One possible solution is to change the calculate_reppoe_threshold function so that the threshold_factor is not used when the difficulty is greater than the minimum:

```rust
std::cmp::min(
    max / u64::from(minimum_difficulty),
    (max / total_active_rep).saturating_mul(my_eligibility).saturating_mul(factor),
)
```

But this has the problem that a data request with 100 witnesses will have the same number of commits than a data request with 5 witnesses.

**Comment by tmpolaczyk** [comment]
This is being implemented in #1976
